tests/config/test_tls.py

```diff
@@ -110,48 +117,48 @@ def test_tls_client_minimum_1_point_3_exists(self):
             self.addCleanup(lambda: delattr(SSL, "OP_NO_TLSv1_3"))
             assert hasattr(SSL, "OP_NO_TLSv1_3")
 
-        config = {"federation_client_minimum_tls_version": 1.3}
+        config: JsonDict = {"federation_client_minimum_tls_version": 1.3}
         t = TestConfig()
         t.tls.read_config(config, config_dir_path="", data_dir_path="")
         self.assertEqual(t.tls.federation_client_minimum_tls_version, "1.3")
 
-    def test_tls_client_minimum_set_passed_through_1_2(self):
+    def test_tls_client_minimum_set_passed_through_1_2(self) -> None:
         """
         The configured TLS version is correctly configured by the ContextFactory.
         """
-        config = {"federation_client_minimum_tls_version": 1.2}
+        config: JsonDict = {"federation_client_minimum_tls_version": 1.2}
         t = TestConfig()
         t.tls.read_config(config, config_dir_path="", data_dir_path="")
 
-        cf = FederationPolicyForHTTPS(t)
+        cf = FederationPolicyForHTTPS(cast(HomeServerConfig, t))
```

Comment:
`TestConfig` implements enough of `HomeServerConfig` for this test, maybe would be clearer if we passed in a `TlsConfig` instead?

Reply:
As in changing FederationPolicyForHTTPS to accept a TlsConfig? Looks like it never uses any other config section, so SveryGTM!

Reply:
Ah, the `TlsConfig` depends on the `server` config though, so we can't do this.

Reply:
Ahh: https://github.com/matrix-org/synapse/blob/493c2fc44abcf3457953cc2f6f23509ff7855253/synapse/config/tls.py#L37-L47

I guess config sections aren't independent :(

(I can feel an Opinion brewing here but I'm going to refrain :P)

Reply:
They *mostly* are, but they are not fully. It is awful.